app/controllers/users/passwords_controller.rb

```diff
@@ -21,13 +21,35 @@ def create
       redirect_to root_path and return
     end
 
-    super do |user|
-      if user.nil? || user.new_record?
-        flash.now[:notice] = ts("We couldn't find an account with that email address or username. Please try again?")
-      else
-        user.update_password_resets_requested
-        user.save
+    if user.present? && !user.new_record?
+      user.update_password_resets_requested
+      user.save
+    end
+
+    super do |target_user|
+      if target_user.nil? || target_user.new_record?
```

Comment:
I wonder if we could do a return early on in the method if user is nil so that we no longer need to use `user&.` in the lines above this.

Reply:
It does change the behaviour ever so slightly (if a non-existent user, the previous method would show an error on the field in addition to flash) but I _think_ that should be OK?

Reply:
I don't think it's a bad thing considering the text of the field error starts with 

> Sorry! We couldn't save this user because:

which is a little confusing.

Reply:
Agreed!